router/fastdp.go

```diff
@@ -1171,6 +1179,45 @@ func (fastdp *FastDatapath) takeDecoder(lock *fastDatapathLock) *EthernetDecoder
 	return dec
 }
 
+// A checkFlowLoop checks whether the flow is free from the following loops:
+// * in_port == out_port, where in_port is non-vxlan vport;
+// * a packet is sent back to a vxlan tunnel it has been received from.
+func (fastdp *FastDatapath) checkFlowLoop(flow *odp.FlowSpec) bool {
+	var vxlanKey odp.TunnelAttrs
+
+	inVports := make(map[odp.VportID]struct{})
+	inVxlan := false
+
+	for _, key := range flow.FlowKeys {
+		switch k := key.(type) {
+		case odp.InPortFlowKey:
+			inVports[k.VportID()] = struct{}{}
```

Comment:
Trying to figure out why this is a map - under what circumstances will there be more than one `InPortFlowKey`?